test/index.coffee

```diff
@@ -0,0 +1,89 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+
+gc = require('../lib/index')
+dockerUtils = require('../lib/docker.coffee')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST == '1' || false
+
+promiseToBool = (p) ->
+	p.return(true).catchReturn(false)
+
+pullAsync = (docker, tag) ->
+	new Promise (resolve, reject) ->
+		docker.pull tag, (err, stream) ->
+			return reject(err) if err?
+			stream.on('data', _.noop)
+			stream.once('end', resolve)
+
+dockerUtils.getDocker()
+.then (docker) ->
+	describe 'Garbage collection', ->
+		# This test case is a little weird, it requires that no other images are present on
+		# the system to ensure that the correct one is being removed. Because of this, you
+		# can use the SKIP_GC_TEST env var to inform the test suite not to run this test
+		it 'should remove the LRU image', ->
+			this.timeout(600000)
+			return Promise.resolve() if SKIP_GC_TEST
+			# first pull some images, so we know in which order they are referenced
+
+			# Create the promises with a standard map so that they're done in the correct
+			# order (bluebird doesn't guarantee this)
```

Comment:
So when we garbage collect we want to use the least recently used image, that's what this code is testing. So the docker-event-stream module listens for the docker events, and stores the used time.
Knowing which order that the pulls were started in allows us to be sure that the ones we're removing are the LRU images.